src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+  /**
+   * Create a `Documentation` object for a remote package available in the npm registry.
+   */
+  public static async forRemotePackage(name: string, version: string, options?: DocumentationOptions): Promise<Documentation> {
+    return withTempDir(async (workdir: string) => {
+      const env = {
+        ...process.env,
+        // npm fails with EROFS if $HOME is read-only, even if it won't write there
+        HOME: os.tmpdir(),
+      };
+
+      await exec('npm install npm@7', {
+        cwd: workdir,
+        env,
+      });
+      await exec(`${workdir}/node_modules/.bin/npm install --ignore-scripts --no-bin-links --no-save ${name}@${version}`, {
+        cwd: workdir,
+        env,
+      });
+
+      return Documentation.forLocalPackage(path.join(workdir, 'node_modules', name), { ...options, assembliesDir: workdir } );
+    });
+  }
+
+  /**
+   * Create a `Documentation` object for a package available in the local file system.
+   */
+  public static async forLocalPackage(root: string, options?: FromLocalPackageDocumentationOptions): Promise<Documentation> {
+    const manifestPath = path.join(root, 'package.json');
+    if (!(await fs.pathExists(manifestPath))) {
+      throw new Error(`Unable to locate ${manifestPath}`);
     }
 
-    this.includeApiReference = options.apiReference ?? true;
-    this.includeReadme = options.readme ?? true;
+    // normally the assemblies are located in subdirectories
+    // of the root package dir (i.e ./node_modules)
+    const assembliesDir = options?.assembliesDir ?? root;
+
+    const manifest = JSON.parse(await fs.readFile(manifestPath, 'utf-8'));
+    return Documentation.forAssembly(manifest.name, assembliesDir);
+  }
+
+  /**
+   * Create a `Documentation` object for a specific assembly that resides in a directory of assemblies.
+   * The directory will comprise the entire type-system available to the generartion process.
+   */
+  public static async forAssembly(assemblyName: string, assembliesDir: string, options?: DocumentationOptions): Promise<Documentation> {
+    return withTempDir(async (workdir: string) => {
+
+      // always better not to operate on an externally provided directory
+      await fs.copy(assembliesDir, workdir);
+
+      let transpile, language;
+
+      switch (options?.language ?? 'ts') {
+        case 'python':
+          language = TargetLanguage.PYTHON;
+          transpile = new PythonTranspile();
+          break;
+        case 'ts':
+          transpile = new TypeScriptTranspile();
+          break;
+        default:
+          throw new Error(`Unsupported language: ${options?.language}`);
+      }
+      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
+      return new Documentation(assembly, transpile);
+    });
+
+  }
+
+  private constructor(
+    private readonly assembly: reflect.Assembly,
+    private readonly transpile: Transpile) {
   }
```

Comment:
Style nitpick (this would reduce unrelated changes if one was to ever add more arguments to this constructor)

```suggestion
  private constructor(
    private readonly assembly: reflect.Assembly,
    private readonly transpile: Transpile,
  ) {
  }
```

Reply:
👍